src/settings.js

```diff
@@ -227,7 +290,111 @@ const settings = {
+        if (startedFromHomeScreen){
+            await returnToHomeScreen();
+        }
+    },
+    /**
+     * @summary Defaults that can be used to apply to all settings
+     * @example await settings.Apply(settings.Defaults)
+     */
+    Defaults: {
+        /**
+         * @summary DeliveryLimits: { maxBolus: '10.0', maxBasalRate: '3.0' }
+         */
+        DeliveryLimits: { maxBolus: '10.0', maxBasalRate: '3.0' },
+        /**
+         * @summary BasalRates: [{ time: '12:00 AM', unitsPerHour: '0.1' }]
+         */
+        BasalRates: [{ time: '12:00 AM', unitsPerHour: '0.1' }],
+        /**
+         * @summary SuspendThreshold: { threshold: '75' }
+         */
+        SuspendThreshold: { threshold: '75' },
+        /**
+         * @summary SelectInsulinModel: { value: 2, name: 'Rapid-Acting – Children' }
+         */
+        SelectInsulinModel: { value: 2, name: 'Rapid-Acting – Children' },
+        /**
+         * @summary CarbRatios: [{ time: '12:00 AM', carbGramsPerInsulinUnit: '8' }]
+         */
+        CarbRatios: [{ time: '12:00 AM', carbGramsPerInsulinUnit: '8' }],
+        /**
+         * @summary InsulinSensitivities: [{ time: '12:00 AM', bgValuePerInsulinUnit: '500' }]
+         */
+        InsulinSensitivities: [{ time: '12:00 AM', bgValuePerInsulinUnit: '500' }],
+        /**
+         * @summary  CorrectionRanges: [{ time: '12:00 AM', min: '179', max: '180' }]
+         */
+        CorrectionRanges: [{ time: '12:00 AM', min: '179', max: '180' }],
+        /**
+         * @summary  PreMealCorrectionRange: { min: '179', max: '180' }
+         */
+        PreMealCorrectionRange: { min: '179', max: '180' },
+        /**
+         *  @summary  ClosedLoop: true
+         */
+        ClosedLoop: true,
+    },
+    Type:{
+        BasalRates:'BasalRates',
+        CarbRatios:'CarbRatios',
+        DeliveryLimits:'DeliveryLimits',
+        SelectInsulinModel:'SelectInsulinModel',
+        SuspendThreshold:'SuspendThreshold',
+        InsulinSensitivities:'InsulinSensitivities',
+        CorrectionRanges:'CorrectionRanges',
+        PreMealCorrectionRange:'PreMealCorrectionRange',
+        ClosedLoop:'ClosedLoop',
+    },
+    /**
+     * @summary filter out settings defaults for those that you don't want to apply
+     * @param {object} values
+     * @param {Array} types
+     * @example settings.Filter(settings.Defaults, [settings.Types.BasalRates])
+     * @returns filtered Defaults set
+     */
+    Filter(values,types) {
+        const filtered = values;
+        for (const type of types) {
+            delete filtered[type];
+        }
+        return filtered;
+    },
+    /**
+     * @summary helper function to set seetings by applying the {settings.Defaults}
+     * @param  {Defaults} values list of settings that will not be applied
+     * @example await settings.Apply(settings.Defaults)
+     */
+    async Apply(values) {
+        startedFromHomeScreen = false;
```

Comment:
[nit] The `startedFromHomeScreen` seems a bit "hackish". Consider an alternate solution to "remember" what view you are on. Perhaps a stack that you push and pop views from. Not sure exactly how it would work, but this way seems fragile. Just a thought.

Reply:
yeah I agree ... i am still mulling over the best way to do it. This works for now but will need changing in the near future